src/utils/symlinkPackageDependencies.js

```diff
@@ -14,30 +14,41 @@ import * as yarn from './yarn';
 export default async function symlinkPackageDependencies(
   project: Project,
   pkg: Package,
-  dependencies: Array<string>
+  dependencies: Array<string>,
+  pkgs: Array<Packages>,
+  dependencyGraph,
 ) {
-  let projectDeps = project.pkg.getAllDependencies();
-  let pkgDependencies = project.pkg.getAllDependencies();
-  let packages = await project.getPackages();
-  let {
-    graph: dependencyGraph,
-    valid: dependencyGraphValid
-  } = await project.getDependencyGraph(packages);
+
+  let packages = pkgs || await project.getPackages();
```

Comment:
Whats going on with this `||`... is the type of `pkgs` wrong?

Reply:
No, it supports either passing the `pkgs` in or fetching them. That was mostly a backwards compatible thing, in case this is called from a different flow. I was primarily fixing the install step.

Reply:
Do you want to validate if it is called in any other flows? I don't think it's hard to require people to call `getPackages()` first

Reply:
The only other place I can tell it's called is here...

https://github.com/boltpkg/bolt/blame/master/src/utils/addDependenciesToPackages.js#L100

When I passed the packages and graph in there though, it failed a bunch of tests in the addDependenciesToPackage tests.

Reply:
I'll see if I can dig into why it's failing. Has something to do with the symlinkSpy.

Reply:
I'm now realizing that the only reason the symlinkDependencies function needed the packages was to pass into the graph. So, updating now to only pass the graph into the symlinkDependencies.